### PR TITLE
refactor: change references of `project` to `service`

### PR DESCRIPTION
**Pull request by oddgrd**
Changes references of `project` to `service` to bring the naming up to date with the new architecture in the `feat/deployer` branch.

### CODE REVIEW

File: deployer/Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 name = "shuttle-deployer"
 version = "0.1.0"
 edition = "2021"
-description = "Service with instances created per project for handling the compliation, loading, and execution of Shuttle services"
+description = "Service with instances created per service for handling the compilation, loading, and execution of Shuttle services"
```
Comment (brokad) [inline deployer/Cargo.toml:5]:
```suggestion
description = "Service with instances created per project for handling the compilation, loading, and execution of Shuttle services"
```

File: deployer/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Deployment System
 
-Service that manages the building, loading, and deployment of the Shuttle service(s) that make up a user's Shuttle project.
+Service that manages the building, loading, and deployment of the Shuttle service(s) that make up a user's Shuttle service.
```
Comment (brokad) [inline deployer/README.md:3]:
```suggestion
Service that manages the building, loading, and deployment of the Shuttle service(s) that make up a user's Shuttle project.
```

File: deployer/src/args.rs [hunk 3]
```diff
@@ -2,7 +2,7 @@ use clap::Parser;
 use fqdn::FQDN;
 use shuttle_common::Port;
 
-/// Program to handle the deploys for a single project
+/// Program to handle the deploys for a single service
```
Comment (brokad) [inline deployer/src/args.rs:5]:
```suggestion
/// Program to handle the deploys for a single project
```

**Review by brokad — COMMENTED** [review]
The idea we had is that, within a *project*, a user can have one or more *services*. Exactly one of those *services* is exposed over HTTP at `${project_name}.shuttleapp.rs` - let's call it the *http service* of the user's *project*.

This means we still need `ProjectName` because projects still exist and the url in fact corresponds to the project name.

Then the idea is that, by default, the first deployment that a user makes under a new project is the *http service* of that project, and carries the same name as the project itself. So if we want to be pedantic we can have `type ServiceName = ProjectName` and replace where appropriate in the deployer.

TL;DR: We need to keep `ProjectName` and related `shuttle_common` items. We just need to change names and types, inside the deployer, of things that relate to `shuttle_service::Service` implementations, from *project* to *service*.

I've only commented the diff on a couple of those, because I figured it was easier to explain in the comment than inline in the review! I realise this was not super clear a priori, sorry :smile:

**Comment by oddgrd** [comment]
Ah, I see! I'll update it, thanks for the review. :smiley:

File: deployer/src/deployment/provisioner_factory.rs [hunk 4]
```diff
@@ -101,7 +101,7 @@ impl<R: ResourceRecorder, S: SecretGetter> Factory for ProvisionerFactory<R, S>
         let db_type: DbType = db_type.into();
 
         let request = Request::new(DatabaseRequest {
-            project_name: self.project_name.to_string(),
+            project_name: self.service_name.to_string(),
```
Comment (oddgrd) [inline deployer/src/deployment/provisioner_factory.rs:104]: Renaming this field means we will also need to rename it in the protobuffer and `provisioner`, not sure if that is what we want.

Comment (brokad) [reply deployer/src/deployment/provisioner_factory.rs:104]: I agree with you, we shouldn't rename this.

**Review by brokad — APPROVED** [review]
LGTM 😄
